tools/v2benchmark/etcd_load.go

```diff
@@ -0,0 +1,540 @@
+    "code.google.com/p/go.crypto/ssh"
+    "github.com/coreos/go-etcd/etcd"
+ 
+)
+
+/*
+Declarations :::: 
+    actions     : for passing other functions as arguments to handler function.
+    operation   : which operation to perform.
+    keycount    : number of keys to be added, retrieved, deleted or updated.
+    threads     : number of total threads.
+    pct         : each entry represents percentage of values lying in .
+                  (value_range[i],value_range[i+1]) . 
+                  Example -- pct={10,10,50,30}, for value_range={0,16,32,64,128} .
+    pct_count   : distribution of keys according to pct .
+                  Example -- {100,100,500,300} for keycount=1000 .
+    value_range : value range for keys .
+    results     : result struct from report.go .
+    threads     : Maximum number of threads allowed to run.
+    threads_dist: Distribution of threads according to load
+*/
+
+type actions func(int,int)
+
+
+var (
+    wg sync.WaitGroup
+    operation, conf_file string
+    etcdmem_s, etcdmem_e int
+    pct, pct_count, value_range, thread_dist []int
+    client *etcd.Client
+    start time.Time
+    f *os.File
+    err error
+    key ssh.Signer
+    config *ssh.ClientConfig
+    ssh_client *ssh.Client
+    session *ssh.Session
+    mem_flag, remote_flag bool
+    results chan *result
+)
+
+// Config variables
+var (
+    log_file, etcdhost, etcdport, remote_host, ssh_port,remote_host_user string
+    keycount, operation_count, threads int
+)
+
+// Flag variables are descirbed here .
+var (
+    fhost, fport, foperation, flog_file, fcfg_file *string
+    fkeycount, foperation_count *int
+    fremote_flag, fmem_flag, fhelp *bool
+)
+
+ func init() {
+    // All the defaults mentioned here refer to etcd_load.cfg values.
+    fhelp = flag.Bool("help", false, "shows how to use flags")    
+    fhost = flag.String("h", "null", "etcd instance address."+
+                        "Default=127.0.0.1 from config file")
+    fport = flag.String("p", "null", 
+                        "port on which etcd is running. Defalt=4001")
+    foperation = flag.String("o", "null", 
+                        "operation - create/delete/get/update. Default:create")
+    fkeycount = flag.Int("k",-1,"number of keys involved in operation,"+
+                        "useful for create. Default:100")
+    foperation_count = flag.Int("oc",-1,"number of operations to be performed,"+
+                        " Default:200")
+    flog_file = flag.String("log","null", "logfile name, default : log")
+    fremote_flag = flag.Bool("remote",false," Must be set true if etcd "+
+                        "instance is remote. Default=false")
+    fmem_flag = flag.Bool("mem",false,"When true, memory info is shown."+
+                        " Default=false")
+    fcfg_file = flag.String("c","null","Input the cfg file. Required")
+ }
+
+
+func main() {
+    flag.Parse()
+
+    if *fhelp {
+        flag.Usage()
+        return
+    }
+    if *fcfg_file != "null"{
```

Comment:
You set the default to "null" when you could have just set it to nil. Then you can simply do:

```
if fcfg_file == nil {
    flag.Usage()
    fmt.Println("Please input the cfg file")
}
```